File: cs50x/week6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_text()
    letters = get_letters(text)
    words = get_words(text)
    sentences = get_sentences(text)
    print(letters, words, sentences)

    # Implement index
    index = round((0.0588 * letters / words * 100) - (0.296 * sentences / words * 100) - 15.8)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print("Grade", index)


def get_text():
    while True:
        try:
            text = get_string("No: ")
            if len(text) > 0:
                break
        except ValueError:
            print("Try again")
    return text


def get_letters(text):
    letters = 0
    for i in range(len(text)):
        if (text[i].isalpha()):
            letters += 1
    return letters


def get_words(text):
    words = 1
    for i in range(len(text)):
        if (text[i].isspace()):
            words += 1
    return words


def get_sentences(text):
    sentences = 0
    for i in range(len(text)):
        if text[i] == '?' or text[i] == '!' or text[i] == '.':
            sentences += 1
    return sentences


main()

File: cs50x/week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //get input from user
    string text = get_string("input here: \n");

    int let = count_letters(text);
    int word = count_words(text);
    int sent = count_sentences(text);

    //get average number of words and convert into float
    float w = word / 100.0;
    float l = let / w;
    float s = sent / w;

    //apply CL formula
    int index = round((0.0588 * l) - (0.296 * s) - 15.8);

    //print results
    //below 1
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    // >16
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    // everything else
    else
    {
        printf("Grade %i\n", index);
    }
}
int count_letters(string text)
{
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1 ;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == 32)
        {
            words += 1;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] ==  33) || (text[i] ==  46) || (text[i] ==  63))
        {
            sentences += 1;
        }
    }
    return sentences;
}